esmvalcore/_task.py

```diff
@@ -682,10 +684,10 @@ def _run_sequential(self) -> None:
         for task in sorted(tasks, key=lambda t: t.priority):
             task.run()
 
-    def _run_parallel(self, max_parallel_tasks: int = None) -> None:
+    def _run_parallel(self, max_parallel_tasks=None):
         """Run tasks in parallel."""
         scheduled = self.flatten()
-        running = {}
+        running: Dict[object, Future] = {}
```

Comment:
I also noticed that this is not checked by mypy, it looks like it only checks functions with type annotations in the function signature?

Reply:
`running: Dict[Union[PreprocessingTask, DiagnosticTask], multiprocessing.pool.AsyncResult] = {}`

Probably better to go for `Dict[Type[BaseTask], Future] = {}` then, since `Future` is more generic than `AsyncResult`.

`mypy` gets really upset when empty lists or dicts are initiated, especially when mixed types are put in. So that's why I had to put this in.

Reply:
> Future is more generic than AsyncResult.

But also wrong, if you look at the code you'll see that the values that are stored in this dict are  `multiprocessing.pool.AsyncResult` objects. If you add a type annotation for the return type of the function `-> None` so this code gets checked by mypy, it reports the following error:
```
$ mypy esmvalcore/_task.py 
esmvalcore/_task.py:713: error: Incompatible types in assignment (expression has type "ApplyResult[Any]", target has type "Future[Any]")
esmvalcore/_task.py:717: error: "Future[Any]" has no attribute "ready"
Found 2 errors in 1 file (checked 1 source file)
```

Reply:
Odd, I don't get that error.

Reply:
Should be fixed now 👍

Reply:
Thanks! I only get the error when I add a type hint to the function definition, i.e. this line:
https://github.com/ESMValGroup/ESMValCore/blob/master/esmvalcore/_task.py#L685
If it doesn't contain any type hints, mypy seems to skip checking the function body.